src/RiotAPILibraries.ts

```diff
@@ -137,31 +127,29 @@ export default class RiotAPILibraries {
 
     async describeAPILibrary(json: GithubAPIStruct): Promise<LibraryDescription> {
 
-        return new Promise<LibraryDescription>(async(resolve, reject) => {
-            const libraryResponse = await fetch(json.download_url);
-            const libraryInfo: APILibraryStruct = await libraryResponse.json();
-    
-            if (!libraryInfo.tags || libraryInfo.tags.indexOf("v3") === -1) {
-                resolve({ stars: 0, valid: false, library: null, links: [] });
-            }
-            
-            const repoResponsePromise = fetch(`https://api.github.com/repos/${libraryInfo.owner}/${libraryInfo.repo}`, this.fetchSettings);
-
-            // Make a list of the links
-            let githubLink = `github.com/${libraryInfo.owner}/${libraryInfo.repo}`;
-            let links = libraryInfo.links ? libraryInfo.links.map(link => `[${link.name}](${link.url})`) : []; // Can be empty array or null, sigh
-            if (links.length == 0 || links.some(l => l.indexOf(githubLink) != 0)) // Make sure there is at least the github link
-                links = [`[Github](https://${githubLink})`].concat(links);
-
-            const repoResponse = await repoResponsePromise;
-            const repoInfo = await repoResponse.json();
-
-            resolve({
-                valid: true,
-                stars: repoInfo.stargazers_count,
-                library: libraryInfo,
-                links: links
-            });
-        });        
+        const libraryResponse = await fetch(json.download_url);
+        const libraryInfo: APILibraryStruct = await libraryResponse.json();
+
+        if (!libraryInfo.tags || libraryInfo.tags.indexOf("v3") === -1) {
+            return { stars: 0, valid: false, library: null, links: [] };
+        }
+        
+        const repoResponsePromise = fetch(`https://api.github.com/repos/${libraryInfo.owner}/${libraryInfo.repo}`, this.fetchSettings);
+
+        // Make a list of the links
+        let githubLink = `github.com/${libraryInfo.owner}/${libraryInfo.repo}`;
+        let links = libraryInfo.links ? libraryInfo.links.map(link => `[${link.name}](${link.url})`) : []; // Can be empty array or null, sigh
+        if (links.length == 0 || links.some(l => l.indexOf(githubLink) != 0)) // Make sure there is at least the github link
+            links = [`[Github](https://${githubLink})`].concat(links);
```

Comment:
This works fine, but maybe something like `links.splice(1, 0, '[Github](https://${githubLink})')`? That way you can make links const.

Reply:
Hmm, current one is more readable